src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+            gottenPath += `${remainingRooms[j].name} -> `;
+            break;
+          }
+        }
+      }
+    }
+    if (!nextRoomIsValid) return false;
+    remainingRooms.splice(indexToChopOffRemainingRooms, 1);
+  }
+  if (showPath) return gottenPath.slice(0, -3);
+  return true;
+}
+// console.log(validateMuseumPath(valid, true));
+
+/*
+ticketExtraDetails()
+Write a new function called ticketExtraDetails() that takes a string as its argument. If that string matches 
+one of the ticket "extras", provide information about what someone would be able to do based on purchasing 
+that ticket. The format is up to you.
+*/
+
+function ticketExtraDetails(extras) {
+  //edge case - if the extra is not movie education or terrace return error message
+  const validExtras = ["movie", "education", "terrace"];
+
+  if (!Array.isArray(extras) && !validExtras.includes(extras.toLowerCase())) {
+    return "must be a valid extra: 'movie', 'education', or 'terrace'";
+  } else {
+    extras = extras.toLowerCase();
+  }
+
+  //if argument is not an array and data type is string
+  if (!Array.isArray(extras) && typeof extras === "string") {
+    if (extras === "education") return "If you purchase this ticket extra you will gain access to 2 more rooms and see 2 more dinosaurs!";
+    if (extras === "terrace") return "If you purchase this ticket extra you will gain access to 1 more room!";
+    if (extras === "movie") return "If you purchase this ticket extra you will gain access to 3 more rooms and see 2 more dinosaurs and see 1 jurassic park movie!";
+  }
+
+  //purpose of creating object is to create reusable code for if we ever want to add extras to extras
+  const extrasObj = {
+    //ordered list represents access to x more rooms, see x more dinos, and see 1 jurassic park movie
+    education: [2, 2, 0],
+    terrace: [1, 0, 0],
+    movie: [3, 2, 1],
+  };
+
+  //iterate through our array of strings called extras with reduce method
+  const result = extras.reduce(
+    (acc, cVal) => {
+      //if the current element is education
+      if (cVal === "education") {
+        //iterate through accumulator
+        acc.map((el, index) => {
+          //add and then re-assign element to evaluated result of adding extrasObject at education at index
+          acc[index] += extrasObj["education"][index];
+          return el;
+        });
+      } else if (cVal === "terrace") {
+        //iterate through accumulator
+        acc.map((el, index) => {
+          //add and then re-assign element to evaluated result of adding extrasObject at education at index
+          acc[index] += extrasObj["terrace"][index];
+          return el;
+        });
+      } else if (cVal === "movie") {
+        //iterate through accumulator
+        acc.map((el, index) => {
+          //add and then re-assign element to evaluated result of adding extrasObject at education at index
+          acc[index] += extrasObj["movie"][index];
+          return el;
+        });
+      }
```

Comment:
A few problems with this code block.
1. `.map()` returns an array, which you should be using. You're using it as if it was `.forEach()`. While it's possible to do this, you probably shouldn't.
2. I think you should be able to use the `cVal` as the key. So, `extrasObj[cVal][index]`. This would reduce the amount of code you have here.